src/crypto/algorithms/megolm.js

```diff
@@ -956,6 +973,18 @@ MegolmDecryption.prototype.importRoomKey = function(session) {
         session.sender_claimed_keys,
         true,
     ).then(() => {
+        if (this._crypto.backupInfo) {
+            // don't wait for it to complete
+            this._crypto.backupGroupSession(
```

Comment:
If this method is called because we are restoring keys from the current backup session, it looks inefficient to send back up the key to the homeserver.

Reply:
Yeah, at some point we may want to flag keys that came from the backup in the first place (and what version)